tests/networkpolicy_test.go

```diff
@@ -1,173 +1,236 @@
+				waitForNetworkPolicyDeletion(policy)
+			})
+
+			It("[test_id:1511] should fail to reach serverVMI from clientVMI", func() {
+				By("Connect serverVMI from clientVMI")
+				assertICMPPingFail(clientVMI, serverVMI)
+			})
+
+			It("[test_id:1512] should fail to reach clientVMI from serverVMI", func() {
+				By("Connect clientVMI from serverVMI")
+				assertICMPPingFail(serverVMI, clientVMI)
+			})
+			It("[test_id:369] should deny http traffic for ports 80/81 from clientVMI to serverVMI", func() {
+				assertHTTPPingFailed(clientVMI, serverVMI, 80)
+				assertHTTPPingFailed(clientVMI, serverVMI, 81)
+			})
+
 		})
 
-	})
+		Context("and vms limited by allow same namespace networkpolicy", func() {
+			var policy *networkv1.NetworkPolicy
 
-	Context("vms limited by allow same namespace networkpolicy", func() {
-		var policy *v1network.NetworkPolicy
-
-		BeforeEach(func() {
-			// allow-same-namespave networkpolicy will only allow the traffice inside the namespace
-			By("Create allow-same-namespace networkpolicy")
-			networkpolicy := &v1network.NetworkPolicy{
-				ObjectMeta: v13.ObjectMeta{
-					Name: "allow-same-namespace",
-				},
-				Spec: v1network.NetworkPolicySpec{
-					PodSelector: v13.LabelSelector{},
-					Ingress: []v1network.NetworkPolicyIngressRule{
+			BeforeEach(func() {
+				// allow-same-namespace networkpolicy will only allow the traffic inside the namespace
+				By("Create allow-same-namespace networkpolicy")
+				policy = createNetworkPolicy(serverVMI.Namespace, "allow-same-namespace", metav1.LabelSelector{},
+					[]networkv1.NetworkPolicyIngressRule{
 						{
-							From: []v1network.NetworkPolicyPeer{
+							From: []networkv1.NetworkPolicyPeer{
 								{
-									PodSelector: &v13.LabelSelector{},
+									PodSelector: &metav1.LabelSelector{},
 								},
 							},
 						},
 					},
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
-		})
+				)
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+			})
 
-		It("[test_id:1513] should be successful to reach vmia from vmib", func() {
-			By("Connect vmia from vmib in same namespace")
-			assertPingSucceed(vmib, vmia)
-		})
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
+
+			It("[test_id:1513] should succeed pinging between two VMI/s in the same namespace", func() {
+				assertICMPPingSucceed(clientVMI, serverVMI)
+			})
+
+			It("[test_id:1514] should fail pinging between two VMI/s each on different namespaces", func() {
+				assertICMPPingFail(clientVMIAlternativeNamespace, serverVMI)
+			})
 
-		It("[test_id:1514] should be failed to reach vmia from vmic", func() {
-			By("Connect vmia from vmic in differnet namespace")
-			assertPingFail(vmic, vmia)
 		})
 
-	})
+		Context("and connectivity between VMI/s with label different from networkpolicy labelSelector is blocked", func() {
```

Comment:
how about something like "and ingress traffic to VMI identified via label is blocked" - or something like that ? 

Really sorry to have opened this can of worms ... Best if we sync up w/ @EdDev first.

Reply:
done